src/components/Provider/Provider.tsx

```diff
@@ -99,27 +114,20 @@ class Provider extends React.Component<any, any> {
   }
 
   render() {
-    const { componentVariables, siteVariables, children, rtl } = this.props
-    const renderer = rtl ? felaRtlRenderer : felaLtrRenderer
-
-    // ensure we don't assign `undefined` values to the theme context
-    // they will override values down stream
-    const theme: any = {
-      renderer,
-      rtl: !!rtl,
-    }
-
-    if (siteVariables) {
-      theme.siteVariables = siteVariables
-    }
-    if (componentVariables) {
-      theme.componentVariables = componentVariables
-    }
+    const { theme, children } = this.props
 
     return (
-      <RendererProvider renderer={renderer}>
-        <ThemeProvider theme={theme}>{children}</ThemeProvider>
-      </RendererProvider>
+      <ProviderConsumer
+        render={(incomingTheme: IThemePrepared) => {
+          const outgoingTheme: IThemePrepared = mergeThemes(incomingTheme, theme)
+
+          return (
+            <RendererProvider renderer={outgoingTheme.renderer}>
+              <ThemeProvider theme={outgoingTheme}>{children}</ThemeProvider>
+            </RendererProvider>
+          )
```

Comment:
This is important.  Each Provider now:

1. consumes the theme from context
1. merges it with its theme prop
1. passes the merged theme downstream

This enables nested cascading themes.